### ISSUE TITLE
Self Hosted: Adding Webauthn FIDO2 does not work. 

### ISSUE THREAD

**Issue by joshuabeny1999** [bitwarden/server#1787]
### Steps To Reproduce

1. Go to Setting - 2FA
2. Manage next to Webauthn
3. Add Security Key via Add
4. Click on Save

### Expected Result

That I could add the Yubikey to activate 2FA login

### Actual Result

I can not save the newly Added Security Key to use for Bitwarden. I have premium over the Family Organisation which is added. Also happens when i try to add a Yubikey via the Yubico Method. 

### Screenshots or Videos

![image](https://user-images.githubusercontent.com/55023662/147559076-97137ca5-058b-4725-b7d8-61d0af958d11.png)
![image](https://user-images.githubusercontent.com/55023662/147559115-021d491f-71f1-4762-bf6a-b843062d13b8.png)

### Additional Context

I use Bitwarden behind an Nginx-Proxy. So I have added following Docker-Compose Overwrite:
```
version: '3'

services:
  nginx:
      environment:
        VIRTUAL_PORT: 8080
        VIRTUAL_HOST: vault.joshuah.ch
        LETSENCRYPT_HOST: vault.joshuah.ch
        LETSENCRYPT_EMAIL: joshua@joshuah.ch

networks:
  default:
    external:
      name: nginx-proxy
```

### Build Version

1.43.5, Web 2.25.0

### Environment

Self-Hosted

### Environment Details

- Windows 10, Webvault in Microsoft Edge Browser

**Comment by SergeantConfused** [comment]
@joshuabeny1999,

Did you configure your server's environment variables? Please read this: https://bitwarden.com/help/article/environment-variables/ and let me know if you're able to enrol a YubiKey as a YubiKey OTP Security Key into your local vault then.


Thank you in advance,

**Comment by joshuabeny1999** [comment]
Hi @SergeantConfused which keys i need that i can authenticate with Webauthn. I do not want to use the Yubikey OTP and prefer the use of Webauthn, which is not cloud based. For that i do not need any Yubico API Keys. Are i'm wrong?

**Comment by joshuabeny1999** [comment]
Update: I also added now the Yubico API Keys. But still with the Keys added it does not work. But my prefered Variant is not Yubico. I want to use Webauthn. 

Here the logfile from the Browser console:
[vault.joshuah.ch-1640714027843.log](https://github.com/bitwarden/server/files/7785847/vault.joshuah.ch-1640714027843.log)

**Comment by phillipmcmahon** [comment]
I am experiencing this exact same issue. All other features of my local install are working correcting. However when trying to add my keys using FIDO2 I see in the logs that the data is exchanged and then when going to what I presume is commit that I get the same error as you showed in the attached logs.

Root error shown in the browser console is

PUT https://xxxxxxx/api/two-factor/webauthn 500

I am not using SSL via the docker setup, instead using an nginx instance to handle SSL etc.

Super basic config, but maybe an issue there?

`server {
    listen          443 ssl http2;
    server_name     xxxxxx;

    include /etc/nginx/conf.d/common.conf;

    location / {
        proxy_pass http://192.168.68.19;
    }

    proxy_set_header    Host                $host;
    proxy_set_header    X-Real-IP           $remote_addr;
    proxy_set_header    X-Forwarded-Host    $server_name;
    proxy_set_header    X-Forwarded-For     $proxy_add_x_forwarded_for;

    proxy_set_header    X-Forwarded-Proto   $scheme;
    proxy_set_header    X-Forwarded-Ssl     on;

    proxy_http_version  1.1;
    proxy_set_header    Upgrade             $http_upgrade;
    proxy_set_header    Connection          "upgrade";

}`

**Comment by phillipmcmahon** [comment]
** update - I performed a standard install via docker using the inbuilt nginx config directly. Can confirm I was able to correctly add a FIDO key.

Seems this is related to using an external nginx instance.

Does anyone have a known good working version that has been tested with adding a FIDO key?

**Comment by joshuabeny1999** [comment]
Thanks for your reply. Good to hear that I am not the only one. Found also a community post. Sadly with no answer. I have some other websites on the same server because I use the jwilder nginx proxy. So using bit arden without it won't work. The onl workaround is to use still TOP with the yubico authenticator for on my Yubico 5C

**Comment by phillipmcmahon** [comment]
I am using that as a workaround for now. However, I have every other option possible except FIDO2 working including DUO with a yubikey registered.

Can you share your nginx conf file?

**Comment by Hinton** [comment]
Hi,

Do you have any further error message from the server logs? The server web vault url needs to configured to match the public url.
I believe the settings should be `globalSettings__baseServiceUri__vault`.

**Comment by phillipmcmahon** [comment]
> Thanks for your reply. Good to hear that I am not the only one. Found also a community post. Sadly with no answer. I have some other websites on the same server because I use the jwilder nginx proxy. So using bit arden without it won't work. The onl workaround is to use still TOP with the yubico authenticator for on my Yubico 5C

@joshuabeny1999 Can you share your nginx file so I can compare any differences?

**Comment by joshuabeny1999** [comment]
@phillipmcmahon I do not have a special nginx config. I am using the https://github.com/nginx-proxy/nginx-proxy with following Template: https://gist.github.com/joshuabeny1999/8ab444f243c2ab7a1b8cdc039d87b875

@Hinton There wever serveral globalSettings__baseServiceUri_* in the env file. Some with links to the api, etc. What i see that all where http:// than https://. I updated them. Also in the bwdata/config.yml the URL was just http. I did afterwards a `bash bitwarden.sh rebuild` and `bash bitwarden.sh start`

Also updated the nginx-proxy to the latest version and then i tried again adding the keys. And it worked. 😄

So it was just a config issue....
